test/acquisition/test_knowledge_gradient.py

```diff
@@ -213,6 +214,53 @@ def test_evaluate_q_knowledge_gradient(self):
                     val_expected = (mean * weights).sum(-1).mean(0)
                     self.assertTrue(torch.allclose(val, val_expected))
 
+    def test_evaluate_KG(self):
+        options = {
+            "num_restarts": 2,
+            "raw_samples": 4,
+            "num_inner_restarts": 2,
+            "raw_inner_samples": 3,
+        }
+        for (
+            d,
+            q,
+            n,
+            n_f,
+            objective,
+            X_pending,
+            current_value,
+            dtype,
+        ) in itertools.product(
```

Comment:
this runs 256 different combinations which is where a lot of the long test times come from. Are some of these test combinations orthogonal and don't add value so we could prune some here?

Reply:
I reduced this to only 2 combinations that likely covers everything I tested for. I may add some more based on codecov report. We could also add additional combinations using mock models as needed